src/main/scala/workflow/Node.scala

```diff
@@ -5,8 +5,16 @@ import org.apache.spark.rdd.RDD
 sealed trait Instruction {
   def getDependencies: Seq[Int]
   def mapDependencies(func: Int => Int): Instruction
+  def execute(deps: Seq[InstructionOutput]): InstructionOutput
 }
 
+sealed trait InstructionOutput
+abstract class DataOutput extends InstructionOutput
+case class RDDOutput(rdd: RDD[_]) extends DataOutput
+case class SingleDataOutput(data: Any) extends DataOutput
+case class EstimatorOutput(estimatorNode: EstimatorNode) extends InstructionOutput
+case class TransformerOutput(transformerNode: TransformerNode) extends InstructionOutput
+
 /**
  * An Instruction that represents the application of a transformer to data inputs.
```

Comment:
Sorry I guess this comment isn't clear enough. I just mean TransformerApplyNode is an instruction that represents the application of a transformer to data inputs (and in turn outputs data).